### PR TITLE
Cublas extension

### PR DESCRIPTION
**Pull request by akochepasov**
Added BLAS 2 and 3 functions (SYRK and SBMV) for real and complex types

### CODE REVIEW

**Comment by akochepasov** [comment]
The respective issue [link](https://github.com/cupy/cupy/issues/6000) to link

**Comment by akochepasov** [comment]
Currently, I cannot find what is the error in rocm-build

File: cupy_backends/hip/cupy_hipblas.h [hunk 1]
```diff
@@ -342,6 +342,22 @@ cublasStatus_t cublasZgerc(cublasHandle_t handle, int m, int n, cuDoubleComplex*
                         reinterpret_cast<hipblasDoubleComplex*>(A), lda);
 }
 
+cublasStatus_t cublasSsbmv(cublasHandle_t handle, cublasFillMode_t uplo, int n, int k,
+                           const float* alpha, const float* A, int lda,
+                           const float* x, int incx,
+                           const float* beta, float* y, int incy) {
+    return hipblasSsbmv(handle, convert_hipblasFillMode_t(uplo), convert_hipblasOperation_t(trans),
```
Comment (kmaehashi) [inline cupy_backends/hip/cupy_hipblas.h:349]:
```suggestion
    return hipblasSsbmv(handle, convert_hipblasFillMode_t(uplo),
```

File: cupy_backends/hip/cupy_hipblas.h [hunk 2]
```diff
@@ -342,6 +342,22 @@ cublasStatus_t cublasZgerc(cublasHandle_t handle, int m, int n, cuDoubleComplex*
...
+                        n, k, alpha, A, lda, x, incx, beta, y, incy);
+}
+
+cublasStatus_t cublasDsbmv(cublasHandle_t handle, cublasFillMode_t uplo, int n, int k,
+                           const double* alpha, const double* A, int lda,
+                           const double* x, int incx,
+                           const double* beta, double* y, int incy) {
+    return hipblasDsbmv(handle, convert_hipblasFillMode_t(uplo), convert_hipblasOperation_t(trans),
```
Comment (kmaehashi) [inline cupy_backends/hip/cupy_hipblas.h:357]:
```suggestion
    return hipblasDsbmv(handle, convert_hipblasFillMode_t(uplo),
```

File: cupy_backends/hip/cupy_hipblas.h [hunk 3]
```diff
@@ -547,6 +563,42 @@ cublasStatus_t cublasZtrsm(cublasHandle_t handle, cublasSideMode_t size, cublasF
...
+                        reinterpret_cast<const hipblasComplex*>(A), lda,
+                        reinterpret_cast<const hipblasComplex*>(beta),
+                        reinterpret_cast<hipblasComplex*>(C), ldc);
+}
+
+cublasStatus_t cublasZsyrk(cublasHandle_t handle, cublasFillMode_t uplo, cublasOperation_t trans, int n, int k,
+                           const cuDoubleComplex* alpha, const cuDoubleComplex* A, int lda,
+                           const cuDoubleComplex* beta, cuDoubleComplex* C, int ldc)
+{
+    return hipblasZsyrk(handle, convert_hipblasOperation_t(transa), convert_hipblasOperation_t(transb), n, k,
```
Comment (kmaehashi) [inline cupy_backends/hip/cupy_hipblas.h:595]:
```suggestion
    return hipblasZsyrk(handle, convert_hipblasFillMode_t(uplo), convert_hipblasOperation_t(trans), n, k,
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]
:+1: Thanks! This should fix ROCm build errors.

**Comment by akochepasov** [comment]
After some sleep, I just found the same build error reason too. The build and tests passed.

**Review by akochepasov — COMMENTED** [review]
Those changes have been concurrently fixed.

File: cupy_backends/hip/cupy_hipblas.h (hunk 1 above)
Comment (akochepasov) [reply cupy_backends/hip/cupy_hipblas.h]: Those changes have been concurrently implemented. I cannot change the status of the issue. It's still in "changes requested"

**Comment by akochepasov** [comment]
I cannot change the status of the issue from "Changes requested". The changes have been implemented and pushed to the repository.

**Comment by emcastillo** [comment]
> What are the next steps?

The PR looks complete to me (I hadn't reviewed it in depth). So if CIs pass and everything looks good we should merge it soon :)
Thanks!!

**Comment by akochepasov** [comment]
I wanted to make sure I don't miss any important steps. Thanks for the update.

**Review by emcastillo — APPROVED** [review]
LGTM!

**Pull request merged**